### PR TITLE
BREAKING CHANGE(utils): remove humanDate + isoDate, use W3CDate 

### LINKED ISSUE #3981: Human Readable Date Inconsistant And Not Localized
## Important info

- ReSpec version: 28.2.8

## Description of problem

grafik

The dates in the top part of the document are 
1. inconsistant
The first date uses the English name of the month (March) while the second date uses the German name (Januar = January).
Both should use the German name, since the language is set to German.

2. not localized
The correct localization in German would be
`1. März 2022`  and `27. Januar 2022`

The leading `0` is not used in texts and the `.` after the day is mandatory.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #3981

Fixes localization of modificationDate + moved W3CDate to util. 

(yeah, I should have done this in 2 PRs... but no time :( )

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Moved part of this into https://github.com/w3c/respec/pull/4010

**Comment by marcoscaceres** [comment]
Ok, marking as breaking change... removed `utils.isoDate()` (not used) and `utils.humanDate()` (not localized properly). 

Now use `W3CDate` everywhere.

File: src/core/templates/show-people.js [hunk 1]
```diff
@@ -83,7 +81,7 @@ export default function showPeople(conf, propName) {
  * @param {Person} person
  */
 function personToHTML(person) {
-  const l10n = localizationStrings[lang];
+  const l10n = getIntlData(localizationStrings);
```
Comment (sidvishnoi) [inline src/core/templates/show-people.js:84]: Should move this out of this function.

File: src/core/utils.js [hunk 2]
```diff
@@ -29,6 +29,18 @@ export const ISODate = new Intl.DateTimeFormat(["en-ca-iso8601"], {
   day: "2-digit",
 });
 
+// We use an "Australian Date" because it omits the ","
+// after the day of the month, which is required by the W3C.
+const dateLang =
+  docLang === "en" || docLang.startsWith("en-") ? "en-AU" : docLang;
+
+export const W3CDate = new Intl.DateTimeFormat(dateLang, {
```
Comment (sidvishnoi) [inline src/core/utils.js:37]: As we're calling .format everywhere, maybe we should export a function formatDateW3C that accepts a string|Date instead?

File: tests/spec/core/utils-spec.js [hunk 3]
```diff
@@ -239,32 +239,6 @@ describe("Core - Utils", () => {
     });
   });
 
-  describe("humanDate", () => {
```
Comment (sidvishnoi) [inline tests/spec/core/utils-spec.js:242]: Should move W3CDate tests here now that it's part of core/utils?

File: src/core/utils.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/utils.js:37]: Yeah, `.format()` everywhere, but seems like having a wrapper is not necessary. I'd rather just have a single object we work from (checks and conversion are where needed). 

The places where it's doing the wrong thing probably need validateAndRecover()

File: tests/spec/core/utils-spec.js (hunk 3 above)
Comment (marcoscaceres) [reply tests/spec/core/utils-spec.js:242]: Probably, yeah.

Comment (sidvishnoi) [reply tests/spec/core/utils-spec.js:242]: We don't have specific tests for W3CDate, so let's ignore this.

**Pull request merged**